daffodil-runtime1/src/main/scala/org/apache/daffodil/infoset/PartialNextElementResolver.scala

```diff
@@ -0,0 +1,310 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.daffodil.infoset
+
+import org.apache.daffodil.processors.ElementRuntimeData
+import org.apache.daffodil.processors.unparsers.UnparseError
+import org.apache.daffodil.util.Maybe
+import org.apache.daffodil.util.Maybe._
+import org.apache.daffodil.xml.NS
+import org.apache.daffodil.xml.StepQName
+import org.apache.daffodil.xml.QNameBase
+import org.apache.daffodil.exceptions.Assert
+import org.apache.daffodil.exceptions.SchemaFileLocation
+import org.apache.daffodil.xml.NamedQName
+import org.apache.daffodil.processors.ElementRuntimeData
+import org.apache.daffodil.util.MStackOf
+import org.apache.daffodil.processors.TermRuntimeData
+import org.apache.daffodil.util.ResettableIterator
+import scala.language.implicitConversions
+import org.apache.daffodil.processors.UnexpectedElementErrorERD
+import org.apache.daffodil.processors.NamespaceAmbiguousElementErrorERD
+import org.apache.daffodil.processors.ErrorERD
+
+/**
+ * Mixin for InfosetInputter
+ */
+trait NextElementResolver { self: InfosetInputter =>
+
+  /**
+   * Resolves the next element, or fails with an Unparse Error if there is
+   * no mapping.
+   */
+  final def nextElement(name: String, nameSpace: String, hasNamespace: Boolean): ElementRuntimeData = {
+    val iter = trdStack.iter
+    iter.reset()
+    var firstOne: Boolean = true
+    var maybeERD: Maybe[ElementRuntimeData] = Nope
+    import util.control.Breaks._
+
+    breakable {
```

Comment:
I think I've read break should be avoided because it's really slow, like backed by heavy exceptiosn or something?

I *think* you can just put the second half of the wile loop inside the default case or your trd match and you don't need the break? Might be someother way to adjust it to get rid of the breakable.

Reply:
removedd